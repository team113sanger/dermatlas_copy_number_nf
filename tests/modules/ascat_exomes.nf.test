nextflow_process {

    name "Test Process RUN_ASCAT_EXOMES"
    script "modules/ascat.nf"
    process "RUN_ASCAT_EXOMES"

    test("Should run without failures") {
        when {
        params {
            load("$baseDir/tests/testdata/test_params.json")
        }
        process {
        """
        input[0] = tuple([pair_id: "PD54403a_PD54403b", normal: "PD54403b", tumor: "PD54403a", Sex: "M", Phenotype: "N", sexchr: "XY"], 
            file("/lustre/scratch124/jb63/ascat/bams/PD54403b/PD54403b.sample.dupmarked.bam"),
            file("/lustre/scratch124/jb63/ascat/bams/PD54403b/PD54403b.sample.dupmarked.bam.bai"),
            file("/lustre/scratch124/jb63/ascat/bams/PD54403a/PD54403a.sample.dupmarked.bam"), 
            file("/lustre/scratch124/jb63/ascat/bams/PD54403a/PD54403a.sample.dupmarked.bam.bai")
            )
        input[1] = "${params.OUTDIR}"
        input[2] = file("${params.reference_genome}")
        input[3] = file("${params.bait_set}") 
        input[4] = file("${params.resource_files}")
        input[5] = file("${params.gc_file}")
        input[6] = file("${params.rt_file}")
        """
        }
        }

        then {
            assert process.success
            assert snapshot(process.out.plots,process.out.allelefreqs).match()
        }

        
}
}

