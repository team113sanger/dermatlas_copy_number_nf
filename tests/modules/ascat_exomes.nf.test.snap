{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "QC_PD54403b_PD54403a.tsv:md5,ce09d2aa68cd61f0129d40966fdd787c"
                    ]
                ],
                "1": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "ASCAT_estimates_PD54403b_PD54403a.tsv:md5,a6a11dcbc68bf08b6d2129aa4f984786"
                    ]
                ],
                "2": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "gistic2_segs_PD54403b_PD54403a.tsv:md5,57de32da04defbabd34b19dbba27d9ca"
                    ]
                ],
                "3": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        [
                            "After_correction_PD54403b.germline.png:md5,3a23a77b7da89dcb66060800ad87a7d9",
                            "After_correction_PD54403b.tumour.png:md5,371a1a8094c4276ee36b23966d211ff6",
                            "Before_correction_PD54403b.germline.png:md5,3a23a77b7da89dcb66060800ad87a7d9",
                            "Before_correction_PD54403b.tumour.png:md5,692b4d1cdaa5dcc7df34154145ddd4bd",
                            "PD54403b.ASCATprofile.png:md5,e6a910c92d1a1b71d8c8c6f7668fb971",
                            "PD54403b.ASPCF.png:md5,21b1eda500fb8c39329952e7d2d3bef5",
                            "PD54403b.rawprofile.png:md5,bc8794e8e67757a721950f75581b4e8d",
                            "PD54403b.sunrise.png:md5,3550f9436dc8498dc55f29872035b42b"
                        ]
                    ]
                ],
                "4": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "ASCAT_objects.Rdata:md5,7e2ac5e2c05330b5b33d6beae168f7b5"
                    ]
                ],
                "5": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        [
                            "PD54403a_alleleFrequencies_chr1.txt:md5,8ea5e50bceb965239f21ddf66b5f5420",
                            "PD54403a_alleleFrequencies_chr10.txt:md5,488bdf098dda334e8ad57cf5aecae8a5",
                            "PD54403a_alleleFrequencies_chr11.txt:md5,3747e72b39f6bd30955ad6d609c8a105",
                            "PD54403a_alleleFrequencies_chr12.txt:md5,272d0f23b303071c649673426c243510",
                            "PD54403a_alleleFrequencies_chr13.txt:md5,32bc3ca544624814d65249b2edd99e2d",
                            "PD54403a_alleleFrequencies_chr14.txt:md5,5b3426a331c72b6c3fb26a112d0788a9",
                            "PD54403a_alleleFrequencies_chr15.txt:md5,0dde47a6acfeb77964a98b7c067b2d41",
                            "PD54403a_alleleFrequencies_chr16.txt:md5,0537cc135862087c0c686cc0b5ec6bf4",
                            "PD54403a_alleleFrequencies_chr17.txt:md5,6a32fb2ee28e56fc1c2ddca94dc3f58a",
                            "PD54403a_alleleFrequencies_chr18.txt:md5,15f9a40775184a994a754aeecdfaaa50",
                            "PD54403a_alleleFrequencies_chr19.txt:md5,2c6decebc066ae079e4a06977cfc9a1b",
                            "PD54403a_alleleFrequencies_chr2.txt:md5,797cb03d98b190c72b3ec95f9ccba0ff",
                            "PD54403a_alleleFrequencies_chr20.txt:md5,94fbd14930403c52fda97987f973644c",
                            "PD54403a_alleleFrequencies_chr21.txt:md5,1ead93e6da158fd535cdb537545a47d2",
                            "PD54403a_alleleFrequencies_chr22.txt:md5,114793e7c52c9cb9c3c36408a59eadf2",
                            "PD54403a_alleleFrequencies_chr3.txt:md5,85d70edbde4a913cce3ff6d4ec2994c4",
                            "PD54403a_alleleFrequencies_chr4.txt:md5,17a6eaca6559c9e4f05900b3eff1e2c8",
                            "PD54403a_alleleFrequencies_chr5.txt:md5,8c33442080cc06b8bff22d254233b527",
                            "PD54403a_alleleFrequencies_chr6.txt:md5,d7d40158755e735e62eb6b16be8fef7d",
                            "PD54403a_alleleFrequencies_chr7.txt:md5,46157f7ec93acb06ef2f49a96c76da56",
                            "PD54403a_alleleFrequencies_chr8.txt:md5,0dcb11dac2a045cf172ad2c70ab70289",
                            "PD54403a_alleleFrequencies_chr9.txt:md5,9d7fa9e1a10a08053aea1c870c8a54bd",
                            "PD54403a_alleleFrequencies_chrX.txt:md5,0b6867e5a16781c21a777294b82cb419",
                            "PD54403b_alleleFrequencies_chr1.txt:md5,c4051fd53fe61223b28a9fbcabc7489a",
                            "PD54403b_alleleFrequencies_chr10.txt:md5,d0da8ca115bcf8c36977a383a5c84a16",
                            "PD54403b_alleleFrequencies_chr11.txt:md5,1626a12ba856ed5a4f2dfd7777ac7356",
                            "PD54403b_alleleFrequencies_chr12.txt:md5,91b19fa50309a7e9a655bf08821b1a5a",
                            "PD54403b_alleleFrequencies_chr13.txt:md5,b1d0812583adc1e9fce21751ceda02fb",
                            "PD54403b_alleleFrequencies_chr14.txt:md5,9479d94f17ed057dddc1c4363f742f59",
                            "PD54403b_alleleFrequencies_chr15.txt:md5,9372bb73d68fc4f59caee9669ccf17c7",
                            "PD54403b_alleleFrequencies_chr16.txt:md5,637a590fe9181e685c9945150c063657",
                            "PD54403b_alleleFrequencies_chr17.txt:md5,848ea59e046de84ddc359f8b215c0b35",
                            "PD54403b_alleleFrequencies_chr18.txt:md5,4681cde9bae8e61b5d3a45b26395fd50",
                            "PD54403b_alleleFrequencies_chr19.txt:md5,ddf82a47d3795acc928d1834c45bde1d",
                            "PD54403b_alleleFrequencies_chr2.txt:md5,f42390229bc41d9637ed0a789ed5ca87",
                            "PD54403b_alleleFrequencies_chr20.txt:md5,e47ac4ca15f5958058242672e8dbaae4",
                            "PD54403b_alleleFrequencies_chr21.txt:md5,0ea7df58ba490732bde4b17906435ce3",
                            "PD54403b_alleleFrequencies_chr22.txt:md5,7604c2d97125ddef26e347c0749d1b9d",
                            "PD54403b_alleleFrequencies_chr3.txt:md5,f73ae83bce2d2b560f8d6ac7e52f0b2e",
                            "PD54403b_alleleFrequencies_chr4.txt:md5,b53e8c10981c048c40f88a6d75773ab6",
                            "PD54403b_alleleFrequencies_chr5.txt:md5,cc0ec3ce7d695cc3382f86b67877bd64",
                            "PD54403b_alleleFrequencies_chr6.txt:md5,4c64eeb8ff1d163a3aa9b80e09d87bb7",
                            "PD54403b_alleleFrequencies_chr7.txt:md5,5bda5b98a005e2ef88d71452bac27ec1",
                            "PD54403b_alleleFrequencies_chr8.txt:md5,4b85defc99e91c6c10967ec7b2c55eb2",
                            "PD54403b_alleleFrequencies_chr9.txt:md5,c74dc2c450022849d1d8b6592ac10fd7",
                            "PD54403b_alleleFrequencies_chrX.txt:md5,d464307b327e0c5f0774bd3541782eea"
                        ]
                    ]
                ],
                "6": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        [
                            "PD54403b_normalBAF.txt:md5,b7cc40c6cafba50e0c5f5508b6fa10ed",
                            "PD54403b_normalBAF_rawBAF.txt:md5,c498015d99ca5bb44b99355da52ba5f6",
                            "PD54403b_tumourBAF.txt:md5,a527e8db10b7521d51ee40f757ec9d4a",
                            "PD54403b_tumourBAF_rawBAF.txt:md5,6211b75a830784e4ae58d3a0c85cf0fe"
                        ]
                    ]
                ],
                "7": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        [
                            "PD54403b_normalLogR.txt:md5,dfeaf349869240fc3484eb31794990c8",
                            "PD54403b_tumourLogR.txt:md5,6c288ee98f4deb49d5a0af2cc45b17e2"
                        ]
                    ]
                ],
                "8": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "PD54403b.segments.txt:md5,21397cd1b7cf37532c785939e6b18cc6"
                    ]
                ],
                "allelefreqs": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        [
                            "PD54403a_alleleFrequencies_chr1.txt:md5,8ea5e50bceb965239f21ddf66b5f5420",
                            "PD54403a_alleleFrequencies_chr10.txt:md5,488bdf098dda334e8ad57cf5aecae8a5",
                            "PD54403a_alleleFrequencies_chr11.txt:md5,3747e72b39f6bd30955ad6d609c8a105",
                            "PD54403a_alleleFrequencies_chr12.txt:md5,272d0f23b303071c649673426c243510",
                            "PD54403a_alleleFrequencies_chr13.txt:md5,32bc3ca544624814d65249b2edd99e2d",
                            "PD54403a_alleleFrequencies_chr14.txt:md5,5b3426a331c72b6c3fb26a112d0788a9",
                            "PD54403a_alleleFrequencies_chr15.txt:md5,0dde47a6acfeb77964a98b7c067b2d41",
                            "PD54403a_alleleFrequencies_chr16.txt:md5,0537cc135862087c0c686cc0b5ec6bf4",
                            "PD54403a_alleleFrequencies_chr17.txt:md5,6a32fb2ee28e56fc1c2ddca94dc3f58a",
                            "PD54403a_alleleFrequencies_chr18.txt:md5,15f9a40775184a994a754aeecdfaaa50",
                            "PD54403a_alleleFrequencies_chr19.txt:md5,2c6decebc066ae079e4a06977cfc9a1b",
                            "PD54403a_alleleFrequencies_chr2.txt:md5,797cb03d98b190c72b3ec95f9ccba0ff",
                            "PD54403a_alleleFrequencies_chr20.txt:md5,94fbd14930403c52fda97987f973644c",
                            "PD54403a_alleleFrequencies_chr21.txt:md5,1ead93e6da158fd535cdb537545a47d2",
                            "PD54403a_alleleFrequencies_chr22.txt:md5,114793e7c52c9cb9c3c36408a59eadf2",
                            "PD54403a_alleleFrequencies_chr3.txt:md5,85d70edbde4a913cce3ff6d4ec2994c4",
                            "PD54403a_alleleFrequencies_chr4.txt:md5,17a6eaca6559c9e4f05900b3eff1e2c8",
                            "PD54403a_alleleFrequencies_chr5.txt:md5,8c33442080cc06b8bff22d254233b527",
                            "PD54403a_alleleFrequencies_chr6.txt:md5,d7d40158755e735e62eb6b16be8fef7d",
                            "PD54403a_alleleFrequencies_chr7.txt:md5,46157f7ec93acb06ef2f49a96c76da56",
                            "PD54403a_alleleFrequencies_chr8.txt:md5,0dcb11dac2a045cf172ad2c70ab70289",
                            "PD54403a_alleleFrequencies_chr9.txt:md5,9d7fa9e1a10a08053aea1c870c8a54bd",
                            "PD54403a_alleleFrequencies_chrX.txt:md5,0b6867e5a16781c21a777294b82cb419",
                            "PD54403b_alleleFrequencies_chr1.txt:md5,c4051fd53fe61223b28a9fbcabc7489a",
                            "PD54403b_alleleFrequencies_chr10.txt:md5,d0da8ca115bcf8c36977a383a5c84a16",
                            "PD54403b_alleleFrequencies_chr11.txt:md5,1626a12ba856ed5a4f2dfd7777ac7356",
                            "PD54403b_alleleFrequencies_chr12.txt:md5,91b19fa50309a7e9a655bf08821b1a5a",
                            "PD54403b_alleleFrequencies_chr13.txt:md5,b1d0812583adc1e9fce21751ceda02fb",
                            "PD54403b_alleleFrequencies_chr14.txt:md5,9479d94f17ed057dddc1c4363f742f59",
                            "PD54403b_alleleFrequencies_chr15.txt:md5,9372bb73d68fc4f59caee9669ccf17c7",
                            "PD54403b_alleleFrequencies_chr16.txt:md5,637a590fe9181e685c9945150c063657",
                            "PD54403b_alleleFrequencies_chr17.txt:md5,848ea59e046de84ddc359f8b215c0b35",
                            "PD54403b_alleleFrequencies_chr18.txt:md5,4681cde9bae8e61b5d3a45b26395fd50",
                            "PD54403b_alleleFrequencies_chr19.txt:md5,ddf82a47d3795acc928d1834c45bde1d",
                            "PD54403b_alleleFrequencies_chr2.txt:md5,f42390229bc41d9637ed0a789ed5ca87",
                            "PD54403b_alleleFrequencies_chr20.txt:md5,e47ac4ca15f5958058242672e8dbaae4",
                            "PD54403b_alleleFrequencies_chr21.txt:md5,0ea7df58ba490732bde4b17906435ce3",
                            "PD54403b_alleleFrequencies_chr22.txt:md5,7604c2d97125ddef26e347c0749d1b9d",
                            "PD54403b_alleleFrequencies_chr3.txt:md5,f73ae83bce2d2b560f8d6ac7e52f0b2e",
                            "PD54403b_alleleFrequencies_chr4.txt:md5,b53e8c10981c048c40f88a6d75773ab6",
                            "PD54403b_alleleFrequencies_chr5.txt:md5,cc0ec3ce7d695cc3382f86b67877bd64",
                            "PD54403b_alleleFrequencies_chr6.txt:md5,4c64eeb8ff1d163a3aa9b80e09d87bb7",
                            "PD54403b_alleleFrequencies_chr7.txt:md5,5bda5b98a005e2ef88d71452bac27ec1",
                            "PD54403b_alleleFrequencies_chr8.txt:md5,4b85defc99e91c6c10967ec7b2c55eb2",
                            "PD54403b_alleleFrequencies_chr9.txt:md5,c74dc2c450022849d1d8b6592ac10fd7",
                            "PD54403b_alleleFrequencies_chrX.txt:md5,d464307b327e0c5f0774bd3541782eea"
                        ]
                    ]
                ],
                "bafs": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        [
                            "PD54403b_normalBAF.txt:md5,b7cc40c6cafba50e0c5f5508b6fa10ed",
                            "PD54403b_normalBAF_rawBAF.txt:md5,c498015d99ca5bb44b99355da52ba5f6",
                            "PD54403b_tumourBAF.txt:md5,a527e8db10b7521d51ee40f757ec9d4a",
                            "PD54403b_tumourBAF_rawBAF.txt:md5,6211b75a830784e4ae58d3a0c85cf0fe"
                        ]
                    ]
                ],
                "estimates": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "ASCAT_estimates_PD54403b_PD54403a.tsv:md5,a6a11dcbc68bf08b6d2129aa4f984786"
                    ]
                ],
                "gistic_inputs": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "gistic2_segs_PD54403b_PD54403a.tsv:md5,57de32da04defbabd34b19dbba27d9ca"
                    ]
                ],
                "logrs": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        [
                            "PD54403b_normalLogR.txt:md5,dfeaf349869240fc3484eb31794990c8",
                            "PD54403b_tumourLogR.txt:md5,6c288ee98f4deb49d5a0af2cc45b17e2"
                        ]
                    ]
                ],
                "plots": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        [
                            "After_correction_PD54403b.germline.png:md5,3a23a77b7da89dcb66060800ad87a7d9",
                            "After_correction_PD54403b.tumour.png:md5,371a1a8094c4276ee36b23966d211ff6",
                            "Before_correction_PD54403b.germline.png:md5,3a23a77b7da89dcb66060800ad87a7d9",
                            "Before_correction_PD54403b.tumour.png:md5,692b4d1cdaa5dcc7df34154145ddd4bd",
                            "PD54403b.ASCATprofile.png:md5,e6a910c92d1a1b71d8c8c6f7668fb971",
                            "PD54403b.ASPCF.png:md5,21b1eda500fb8c39329952e7d2d3bef5",
                            "PD54403b.rawprofile.png:md5,bc8794e8e67757a721950f75581b4e8d",
                            "PD54403b.sunrise.png:md5,3550f9436dc8498dc55f29872035b42b"
                        ]
                    ]
                ],
                "qc_metrics": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "QC_PD54403b_PD54403a.tsv:md5,ce09d2aa68cd61f0129d40966fdd787c"
                    ]
                ],
                "rdata": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "ASCAT_objects.Rdata:md5,7e2ac5e2c05330b5b33d6beae168f7b5"
                    ]
                ],
                "segments": [
                    [
                        {
                            "pair_id": "PD54403a_PD54403b",
                            "normal": "PD54403a",
                            "tumor": "PD54403b",
                            "Sex": "M",
                            "Phenotype": "N",
                            "sexchr": "XY"
                        },
                        "PD54403b.segments.txt:md5,21397cd1b7cf37532c785939e6b18cc6"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-28T14:17:44.867181"
    }
}